htdocs/api/v0.0.3-dev/candidates/visits/Dicoms.php

```diff
@@ -0,0 +1,148 @@
+     * @param string $VisitLabel The visit label to be serialized
+     */
+    public function __construct($method, $CandID, $VisitLabel)
+    {
+        $requestDelegationCascade = $this->AutoHandleRequestDelegation;
+
+        $this->AutoHandleRequestDelegation = false;
+
+        if (empty($this->AllowedMethods)) {
+            $this->AllowedMethods = ['GET'];
+        }
+        $this->CandID     = $CandID;
+        $this->VisitLabel = $VisitLabel;
+
+        parent::__construct($method, $CandID, $VisitLabel);
+
+        if ($requestDelegationCascade) {
+            $this->handleRequest();
+        }
+
+    }
+
+    /**
+     * Handles a GET request
+     *
+     * @return none, but populates $this->JSON
+     */
+    public function handleGET()
+    {
+        $this->JSON = [
+                       'Meta' => [
+                                  'CandID' => $this->CandID,
+                                  'Visit'  => $this->VisitLabel,
+                                 ],
+                      ];
+        $this->JSON['DicomTars'] = $this->getVisitDicoms();
+
+    }
+
+    /**
+     * Gets a list of Tars for this visit. Tarname only.
+     *
+     * @return array of strings of Tarnames
+     */
+    function getVisitDicoms()
+    {
+        $factory   = \NDB_Factory::singleton();
+        $DB        = $factory->database();
+        $cand_info = $DB->pselectRow(
+            "SELECT c.PSCID, c.Entity_type
+                FROM candidate c
+                    JOIN session s ON (c.CandID=s.CandID)
+                WHERE c.Active='Y' AND s.Active='Y'
+                AND c.CandID=:CID
+                AND s.Visit_label=:VL",
+            [
+             'CID' => $this->CandID,
+             'VL'  => $this->VisitLabel,
+            ]
+        );
+        if ($cand_info['Entity_type'] == 'Scanner') {
+            $ID            = ":PVL";
+            $params['PVL'] = $this->VisitLabel;
+        } else {
+            $ID = "LOWER(CONCAT(:PPSCID, '_', :PCandID, '_', :PVL, '%'))";
+            $params['PPSCID']  = $cand_info['PSCID'];
+            $params['PCandID'] = $this->CandID;
+            $params['PVL']     = $this->VisitLabel;
+        }
+
+        $query = "SELECT SUBSTRING_INDEX(ArchiveLocation, '/', -1) as Tarname,
+           ts.SeriesDescription as SeriesDescription,
+           ts.SeriesNumber as SeriesNumber,                    
+           ts.EchoTime as EchoTime,                    
+           ts.SeriesUID as SeriesUID      
+           FROM tarchive t                      
+           JOIN tarchive_series ts ON (ts.TarchiveID=t.TarchiveID)                
+           WHERE t.PatientName LIKE $ID            
+           GROUP BY t.TarchiveID, ts.SeriesDescription, ts.SeriesNUmber, 
+           ts.EchoTime, ts.SeriesUID
+           ORDER BY Tarname";
+
+        $rows = $DB->pselect($query, $params);
+
+        $result = array();
+        $entry  = array();
+
+        foreach ($rows as $row) {
```

Comment:
i know you explain this in the docs but i think it would be useful to just briefly remind what this loop is doing
maybe at the start of the loop you can just write out an abbrieviated version of the desired structure i.e,
`[{Tarname: '...',`
  `SeriesInfo: [{...}]`
  ` },`
    ...
`]`

since it is a little bit confusing at face value

Reply:
Okay!